torch/utils/tensorboard/pytorch_graph.py

```diff
@@ -0,0 +1,274 @@
+
+    scope = {}
+    nodes_py = Graph_py()
+    for i, node in enumerate(graph.inputs()):
+        if omit_useless_nodes:
+            if len(node.uses()) == 0:  # number of user of the node (= number of outputs/ fanout)
+                continue
+
+        if i < n_inputs:
+            nodes_py.append(Node_py_IO(node, 'input'))
+        else:
+            nodes_py.append(Node_py_IO(node))  # parameter
+
+    for node in graph.nodes():
+        nodes_py.append(Node_py_OP(node))
+
+    for node in graph.outputs():  # must place last.
+        Node_py_IO(node, 'output')
+    nodes_py.findCommonRoot()
+    nodes_py.populate_namespace_from_OP_to_IO()
+    return nodes_py.to_proto()
+
+
+def graph(model, args, verbose=False, **kwargs):
+    def _optimize_trace(trace, operator_export_type):
+        trace.set_graph(_optimize_graph(trace.graph(), operator_export_type))
+
+    def _optimize_graph(graph, operator_export_type):
+        # torch._C._jit_pass_remove_inplace_ops(graph)
+        # we record now record some ops like ones/zeros
+        # into a trace where we previously recorded constants
+        # use constant prop to maintain our current level of onnx support
+        # without implementing symbolics for all of them
+        torch._C._jit_pass_constant_propagation(graph)
+        torch.onnx.utils._split_tensor_list_constants(graph, graph)
+        # run dce to eliminate dead parts of the graph that might have been
+        # left behind by things like symbolic_override
+        torch._C._jit_pass_dce(graph)
+        torch._C._jit_pass_lint(graph)
+
+        # torch._C._jit_pass_canonicalize_ops(graph)
+        torch._C._jit_pass_lint(graph)
+
+        torch._C._jit_pass_peephole(graph, True)
+        torch._C._jit_pass_lint(graph)
+
+        # onnx only supports tensors, but 1 / 2 = 0.5 and tensor(1) / tensor(2) = 0
+        torch._C._jit_pass_prepare_division_for_onnx(graph)
+        # onnx only supports tensors, so we turn all out number types into tensors
+        torch._C._jit_pass_erase_number_types(graph)
+        # onnx does not support tuples, so try to remove them
+        torch._C._jit_pass_lower_all_tuples(graph)
+        torch._C._jit_pass_peephole(graph, True)
+        torch._C._jit_pass_lint(graph)
+
+        if operator_export_type != torch.onnx.utils.OperatorExportTypes.RAW:
+            graph = torch._C._jit_pass_onnx(graph, operator_export_type)
+            torch._C._jit_pass_lint(graph)
+            # torch._C._jit_pass_onnx_peephole(graph)
+            torch._C._jit_pass_lint(graph)
+        torch._C._jit_pass_dce(graph)
+        torch._C._jit_pass_lint(graph)
+        torch._C._jit_pass_fixup_onnx_loops(graph)
+        torch._C._jit_pass_lint(graph)
+        graph = torch._C._jit_pass_canonicalize(graph)
+        torch._C._jit_pass_lint(graph)
+        return graph
+
+    assert LooseVersion(torch.__version__) >= LooseVersion("1.0.0"),\
+        'This version of tensorboardX requires pytorch>=1.0.0.'
```

Comment:
We should delete this assert now, I think.

Reply:
Deleted.